### PR TITLE
[Android] Add support for launching a game through activity monitor.

### PR DESCRIPTION
**Pull request by Sonicadvance1**
This also allows setting the user directory.
This is mainly for FifoCI in the future.

### CODE REVIEW

**Comment by Sonicadvance1** [comment]
Launching goes something like this.
adb shell am start -n org.dolphinemu.dolphinemu/.activities.MainActivity -e AutoStartFile /sdcard/LuigisMansion.gcz -e UserDir /sdcard/dolphin-emu2

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/StartupHandler.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+		}
+
+		Intent intent = parent.getIntent();
+		Bundle extras = intent.getExtras();
+
+		if (extras != null) {
+			String user_dir = extras.getString("UserDir");
+			String start_file = extras.getString("AutoStartFile");
+
+			if (user_dir != null && user_dir.length() != 0)
```
Comment (lioncash) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/StartupHandler.java]: if (!TextUtils.isEmpty(user_dir))

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/StartupHandler.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+				NativeLibrary.SetUserDirectory(user_dir);
+
+			if (start_file != null && start_file.length() != 0) {
```
Comment (lioncash) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/StartupHandler.java]: if (!TextUtils.isEmpty(start_file))

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/StartupHandler.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import android.content.SharedPreferences;
+import android.os.Bundle;
+import android.preference.PreferenceManager;
+import android.util.Log;
+
+import org.dolphinemu.dolphinemu.NativeLibrary;
+import org.dolphinemu.dolphinemu.activities.EmulationActivity;
+import org.dolphinemu.dolphinemu.services.AssetCopyService;
+
+public class StartupHandler {
```
Comment (lioncash) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/StartupHandler.java]: public final class

**Comment by freelancer42** [comment]
This would also allow games to be launched from frontends such as ARC Browser (I'm the developer of that), which would be highly appreciated. Several of my users have requested support for this.

**Pull request merged**